tests/test_grids_engine_integration.py

```
@@ -0,0 +1,135 @@
+        engine = PandasEngine()
+        result = engine.index_points(df, "lat", "lon", level=10)
+        assert "s2_index" in result.columns
+
+    def test_explicit_grid_h3(self, df):
+        pytest.importorskip("h3")
+        from siege_utilities.engines.dataframe_engine import PandasEngine
+        engine = PandasEngine()
+        result = engine.index_points(df, "lat", "lon", grid="h3", resolution=7)
+        assert "h3_index" in result.columns
+
+    def test_index_polygon(self, df):
```

Ruff (0.15.12)

[warning] Unused method argument: `df`

(ARG002)